src/registries/autofill_rules.ts

```diff
@@ -77,6 +86,53 @@ function calculateIncrementBasedOnGroup(group: number[]) {
   return increment;
 }
 
+function getEqualInterval(intervals: DateIncrement[], key: keyof DateIncrement) {
```

Comment:
I don't have good ideas for the name, but it was kinda hard to understand what the function does exactly. Maybe inline it/add some jsDoc ?